In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('avocado.csv', sep = ',', header = 0, index_col = False, names = None)
df

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0               0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1               1  2015-12-20          1.35      54876.98   674.28   44638.81   
2               2  2015-12-13          0.93     118220.22   794.70  109149.67   
3               3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4               4  2015-11-29          1.28      51039.60   941.48   43838.39   
...           ...         ...           ...           ...      ...        ...   
18244           7  2018-02-04          1.63      17074.83  2046.96    1529.20   
18245           8  2018-01-28          1.71      13888.04  1191.70    3431.50   
18246           9  2018-01-21          1.87      13766.76  1191.92    2452.79   
18247          10  2018-01-14          1.93      16205.22  1527.63    2981.04   
18248          11  2018-01-07          1.62      17489.58  2894.77    2356.13   

         4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0       48.16     8696.87     8603.62       93.25          0.0  conventional   
1       58.33     9505.56     9408.07       97.49          0.0  conventional   
2      130.50     8145.35     8042.21      103.14          0.0  conventional   
3       72.58     5811.16     5677.40      133.76          0.0  conventional   
4       75.78     6183.95     5986.26      197.69          0.0  conventional   
...       ...         ...         ...         ...          ...           ...   
18244    0.00    13498.67    13066.82      431.85          0.0       organic   
18245    0.00     9264.84     8940.04      324.80          0.0       organic   
18246  727.94     9394.11     9351.80       42.31          0.0       organic   
18247  727.01    10969.54    10919.54       50.00          0.0       organic   
18248  224.53    12014.15    11988.14       26.01          0.0       organic   

       year            region  
0      2015            Albany  
1      2015            Albany  
2      2015            Albany  
3      2015            Albany  
4      2015            Albany  
...     ...               ...  
18244  2018  WestTexNewMexico  
18245  2018  WestTexNewMexico  
18246  2018  WestTexNewMexico  
18247  2018  WestTexNewMexico  
18248  2018  WestTexNewMexico  

[18249 rows x 14 columns]

In [3]:
df.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

In [4]:
df.tail()

Unnamed: 0        Date  AveragePrice  Total Volume     4046     4225  \
18244           7  2018-02-04          1.63      17074.83  2046.96  1529.20   
18245           8  2018-01-28          1.71      13888.04  1191.70  3431.50   
18246           9  2018-01-21          1.87      13766.76  1191.92  2452.79   
18247          10  2018-01-14          1.93      16205.22  1527.63  2981.04   
18248          11  2018-01-07          1.62      17489.58  2894.77  2356.13   

         4770  Total Bags  Small Bags  Large Bags  XLarge Bags     type  year  \
18244    0.00    13498.67    13066.82      431.85          0.0  organic  2018   
18245    0.00     9264.84     8940.04      324.80          0.0  organic  2018   
18246  727.94     9394.11     9351.80       42.31          0.0  organic  2018   
18247  727.01    10969.54    10919.54       50.00          0.0  organic  2018   
18248  224.53    12014.15    11988.14       26.01          0.0  organic  2018   

                 region  
18244  WestTexNewMexico  
18245  WestTexNewMexico  
18246  WestTexNewMexico  
18247  WestTexNewMexico  
18248  WestTexNewMexico

In [15]:
df.dtypes

Unnamed: 0        int64
Date             object
AveragePrice    float64
Total Volume    float64
4046            float64
4225            float64
4770            float64
Total Bags      float64
Small Bags      float64
Large Bags      float64
XLarge Bags     float64
type             object
year              int64
region           object
dtype: object

In [17]:
df.shape

(18249, 14)

### Question

1)Which region sold the largest amount of avocado ? \
In this region, where the biggest lot of sold avocado came from (4046, 4225, 4770) ? 

Ans = TotalUS sold the largest amount of avocado, the biggest lot of sold avocado came from lot 4046.

In [27]:
df['Total Volume'].max()

62505646.52

In [26]:
df_max = df[df['Total Volume'] == 62505646.52]
df_max

Unnamed: 0        Date  AveragePrice  Total Volume        4046  \
9097           7  2018-02-04          0.87   62505646.52  21620180.9   

             4225        4770   Total Bags  Small Bags  Large Bags  \
9097  20445501.03  1066830.22  19373134.37  13384586.8  5719096.61   

      XLarge Bags          type  year   region  
9097    269450.96  conventional  2018  TotalUS

2)Which region sold the smallest amount of avocado ? \
In this region, where the biggest lot of sold avocado came from (4046, 4225, 4770) ?

Ans = MiamiFtLauderdale sold the smallest amount of avocado, the biggest lot of sold avocado came from lot 4046 and 4225.

In [22]:
df['Total Volume'].min()

84.56

In [28]:
df_min = df[df['Total Volume'] == 84.56]
df_min

Unnamed: 0        Date  AveragePrice  Total Volume  4046  4225  4770  \
10381           7  2015-11-08          1.59         84.56  3.95  3.95   0.0   

       Total Bags  Small Bags  Large Bags  XLarge Bags     type  year  \
10381       76.66       73.33        3.33          0.0  organic  2015   

                  region  
10381  MiamiFtLauderdale

3) Which region sold the highest price of avocado in average ?

Ans = HartfordSpringfield region sold the highest price of avocado in average = $1.818639.

In [57]:
region_avgPrice = df.groupby('region')[['AveragePrice']].mean().sort_values(by=['AveragePrice'], ascending = False)
region_avgPrice.head(10)

AveragePrice
region                           
HartfordSpringfield      1.818639
SanFrancisco             1.804201
NewYork                  1.727574
Philadelphia             1.632130
Sacramento               1.621568
Charlotte                1.606036
Northeast                1.601923
Albany                   1.561036
Chicago                  1.556775
RaleighGreensboro        1.555118

4) Find the total amount of income (Avg_Price*Total_Volume) of each region.

In [60]:
AvgPrice = df['AveragePrice']

In [61]:
TotalVolume = df['Total Volume']

In [69]:
AvgPrice*TotalVolume

0         85434.7046
1         74083.9230
2        109944.8046
3         85311.5220
4         65330.6880
            ...     
18244     27831.9729
18245     23748.5484
18246     25743.8412
18247     31276.0746
18248     28333.1196
Length: 18249, dtype: float64

5) Let AVOCADO  Average Weight : 4046 => 4 ounces, 4225 => 9 ounces, 4770 => 12 ounces \
    Find the number of sold avocadoes by region ? \
    Which region sold the largest number of avocados ?

6) Normally, the customers buy the avocados by unit or in a bags ?

#### Optional

7) Which month is the best time to sell avocado

8)  Baseon the data in 2015, is it true that the region which higher population can sell more avocados?